content/docs/base-components/ReactiveComponent.md

```diff
@@ -15,50 +15,110 @@ redirect_from:
 
 > How does this work?
 >
-> `ReactiveComponent` is a wrapper component that allows you to connect custom component(s) (passed as children) with the Reactivesearch ecosystem.
+> `ReactiveComponent` supports a defaultQuery prop, which can take any ElasticSearch Query DSL object, and provides a callback onData prop which is called on successful query execution by ElasticSearch and can be used to create a side-effect in your app. There is also a render prop which can be used to render any React UI component of your choice which will have access to the hits and aggregations from the defaultQuery.
 
 
 ### Usage
 
-Let's suppose - we are building an e-commerce store for cars which displays a list of cars of a particular brand on their separate page as `example.com/cars/nissan`. Now, we want all the filters on that page (like pricing, type of car, model, year, etc) to only show the data relevant to the given brand (i.e. `nissan`). In this case, `ReactiveComponent` can be used with `customQuery` to achieve the desired behavior easily.
+Let's suppose - we are building an e-commerce store for cars which displays a list of cars and 5 popular car brands in tabs UI as the user selects a tab the result gets updated with the cars of selected brand. Now if the user wants to select other brand from tab, the results should be updated to new selected brand. In this case, `ReactiveComponent` can be used with `defaultQuery` to achieve the desired behavior easily.
 
-We can then use the given ReactiveComponent to be watched by all the filters (via `react` prop) to avail the desired brand based filtering for all the filters.
+We can then use the given ReactiveComponent to be watched by the result component (via `react` prop) to avail the desired brand based filtering for the results.
 
-Check demo [here](https://codesandbox.io/s/3ylrrr0r5q).
+Check demo [here](https://codesandbox.io/s/7zrj740oj6).
 
 
 ```js
  <ReactiveComponent
-    componentId="CarSensor"
-    customQuery={props => ({
-        query: {
-                term: {
-                    "brand.keyword": "Ford"
-                }
-            }
+    componentId="Brand"
+    showFilter
+    defaultQuery={() => ({
+        aggs: {
+            'brand.keyword': {
+                terms: {
+                    field: 'brand.keyword',
+                    order: {
+                        _count: 'desc',
+                    },
+                    size: 5,
+                },
+            },
+        },
+    })}
+    render={data => {
+        return <TabComponent dataField="brand.keyword" {...data} />;
+    }}
+/>
+```
+
+You can also take advantage of various ReactiveSearch components static method for generating query. So we our basically selecting a single tab so we can use `SingleList` static method to generate query as follow:
```

Comment:
Can we focus only on ReactiveComponent? While this is good to know, it may be confusing. It's easier to directly apply a Query DSL query instead. After all, SingleList just uses a term query.